workbench-core/datasets-ui/src/api/datasets.ts

```diff
@@ -3,16 +3,20 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { httpApiGet } from '@aws/workbench-core-swb-common-ui';
+import { httpApiGet, httpApiPost } from '@aws/workbench-core-swb-common-ui';
 import useSWR from 'swr';
-import { DatasetItem } from '../models/Dataset';
+import { CreateDatasetForm, DatasetItem } from '../models/Dataset';
 
-const useDatasets = (): { datasets: DatasetItem[], areDatasetsLoading: boolean } => {
+const useDatasets = (): { datasets: DatasetItem[]; areDatasetsLoading: boolean } => {
   const { data, isValidating } = useSWR(() => 'datasets', httpApiGet);
   // TODO: Once datasetService methods return response wrapped in a {data: <response>} body, replace the line below with:
   // const datasets: DatasetItem[] = data?.data ?? [];
   const datasets: DatasetItem[] = data || [];
   return { datasets, areDatasetsLoading: isValidating };
 };
 
-export { useDatasets };
+const createDataset = async (dataset: CreateDatasetForm): Promise<void> => {
```

Comment:
Files should not be sent to the server, instead a presigned URL should be requested for each file and sent directly to the requested URL.

Reply:
This part will be taken care with integration work. Current API does not have description (or project ids)